### PR TITLE
IA1-4514_sidebar-automatic-opening

### PR DESCRIPTION
**Pull request by andreyshepilov**

### CODE REVIEW

File: src/annotator/guest.coffee [hunk 1]
```diff
@@ -93,14 +93,15 @@ module.exports = class Guest extends Delegator
...
+  scrollAndHighlightAnnotation: (selector) ->
     for anchor in @anchors when anchor.highlights?
-      if JSON.stringify(incomingSelector) == JSON.stringify(anchor.target.selector)
+      if JSON.stringify(selector) == JSON.stringify(anchor.target.selector)
         @scrollToAnnotation(anchor)
+    @highlightSelected(selector)
 
-    @highlightSelected(incomingSelector)
+  scrollAndHighlightAnnotationHandler: (event) ->
+    incomingSelector = event.detail.selector
```
Comment (sergeyvdovareize) [inline src/annotator/guest.coffee:103]: you don't really need this veriable

File: src/annotator/guest.coffee [hunk 2]
```diff
@@ -93,14 +93,15 @@ module.exports = class Guest extends Delegator
...
-
+  scrollAndHighlightAnnotation: (selector) ->
     for anchor in @anchors when anchor.highlights?
-      if JSON.stringify(incomingSelector) == JSON.stringify(anchor.target.selector)
+      if JSON.stringify(selector) == JSON.stringify(anchor.target.selector)
         @scrollToAnnotation(anchor)
+    @highlightSelected(selector)
 
-    @highlightSelected(incomingSelector)
+  scrollAndHighlightAnnotationHandler: (event) ->
```
Comment (sergeyvdovareize) [inline src/annotator/guest.coffee:102]: to have 2 methods with names like `scrollAndHighlightAnnotation` and `scrollAndHighlightAnnotationHandler` is a bit confusing

Comment (andreyshepilov) [reply src/annotator/guest.coffee:102]: The "handler" method is just a wrapper for main method to pass argument with additional processing. This two methods is just two parts of one feature, that's why names is similar

**Comment by andreyshepilov** [comment]
This improvement will be moved to another task

**Pull request closed without merging**